ZividOpenCV/CaptureUndistortRGB/CaptureUndistortRGB.cpp

```diff
@@ -0,0 +1,139 @@
+/*
+Undistort an RGB image from a ZDF point cloud using Zivid camera intrinsics.
+*/
+
+#include <Zivid/CloudVisualizer.h>
+#include <Zivid/Zivid.h>
+
+#include <opencv2/calib3d/calib3d.hpp>
+#include <opencv2/core/core.hpp>
+#include <opencv2/highgui/highgui.hpp>
+#include <opencv2/imgproc/imgproc.hpp>
+
+#include <iostream>
+
+cv::Mat pointCloudToRGB(const Zivid::PointCloud &);
+std::tuple<cv::Mat, cv::Mat> getCameraIntrinsics(const Zivid::CameraIntrinsics &);
+void displayRGB(const cv::Mat &, const std::string &);
+
+int main()
+{
+    try
+    {
+        Zivid::Application zivid;
+
+        std::cout << "Connecting to the camera" << std::endl;
+        auto camera = zivid.connectCamera();
+
+        std::cout << "Configuring the camera settings" << std::endl;
+        camera << Zivid::Settings::Iris{ 21 } << Zivid::Settings::ExposureTime{ std::chrono::microseconds{ 20000 } }
+               << Zivid::Settings::Gain{ 1 } << Zivid::Settings::Brightness{ 1.0 };
+
+        std::cout << "Capturing a frame" << std::endl;
+        const auto frame = camera.capture();
+
+        std::cout << "Setting up visualization" << std::endl;
+        Zivid::CloudVisualizer vis;
+        zivid.setDefaultComputeDevice(vis.computeDevice());
+
+        std::cout << "Displaying the point cloud" << std::endl;
+        vis.showMaximized();
+        vis.show(frame);
+        vis.resetToFit();
+
+        std::cout << "Running the visualizer. Blocking until the window closes" << std::endl;
+        vis.run();
+
+        std::cout << "Converting ZDF point cloud to OpenCV format" << std::endl;
+
+        const auto pointCloud = frame.getPointCloud();
+        const auto rgb = pointCloudToRGB(pointCloud);
+
+        std::cout << "Undistorting the RGB image" << std::endl;
+
+        const auto cameraIntrinsticsCV = getCameraIntrinsics(camera.intrinsics());
+        const cv::Mat distortionCoefficients = std::get<0>(cameraIntrinsticsCV);
```

Comment:
For consistency, maybe use `auto` for these four variables too?

Reply:
Done, if we assume the same four.